Source/Core/Core/IOS/USB/Emulated/Infinity.cpp

```diff
@@ -0,0 +1,994 @@
+// Copyright 2022 Dolphin Emulator Project
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "Core/IOS/USB/Emulated/Infinity.h"
+
+#include <array>
+#include <map>
+#include <mutex>
+#include <vector>
+
+#include "Common/Crypto/SHA1.h"
+#include "Common/Logging/Log.h"
+#include "Common/StringUtil.h"
+#include "Common/Timer.h"
+#include "Core/Core.h"
+#include "Core/HW/Memmap.h"
+#include "Core/System.h"
+
+namespace IOS::HLE::USB
+{
+const std::map<const char*, const InfinityFigureInfo> list_infinity_figures = {
```

Comment:
Ideally this should be a constexpr array of std::pairs (with the second pair type using a struct that doesn't use an allocating `std::string`, otherwise this uses up heap memory even when the feature isn't being used.

This should be fine anyway, considering the only time this map is used, is to loop over the elements.

Reply:
@deReeperJosh Can you turn this into an array of `std::pair<std::string_view, InfinityFigureInfo>`? As far as I can tell you're never looking up elements based on the key, so there's no point in using an associative container such as std::map.